File: submissions/StudentTwo SurnameTwo - fzampirolli2/2025-08-27-08-39-35/Q1.py
class CentralNotificacoes:
    _instancia = None
    
    def __new__(cls):
        if cls._instancia == None:
            cls._instancia = super().__new__(cls)
            cls._instancia.msg = ["Nova Conexão", "Mensagem recebida"]
            print("Central iniciada")
            
        else:
            print("Central já ativa")
            
        return cls._instancia
        
    def adicionar(self, msg):
        self.msg.append(msg)
        print(f"Nova notificação: {msg}")
    
    def listar(self):
        print("Notificações:")
        for msg in self.msg:
            print(f"{msg}")
            
msg = input()
comando = input()

app = CentralNotificacoes()

if comando == "adicionar":
    app.adicionar(msg)
    
elif comando == "listar":
    app.adicionar(msg)
    app.listar()
    
elif comando == "nova_instancia":
    app = CentralNotificacoes()

File: submissions/StudentOne SurnameOne - fzampirolli/2025-08-27-09-17-50/Q1.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;

/**
 * Implementação de um sistema de notificações com instância única (Singleton).
 */
public class Q1 {

    // Classe responsável por gerenciar notificações
    public static class GerenciadorNotificacoes {

        private static GerenciadorNotificacoes instanciaUnica;
        private final ArrayList<String> filaNotificacoes;

        private GerenciadorNotificacoes() {
            filaNotificacoes = new ArrayList<>();
        }

        // Método estático para acessar a instância única
        public static GerenciadorNotificacoes acessar() {
            if (instanciaUnica == null) {
                instanciaUnica = new GerenciadorNotificacoes();
                System.out.println("Sistema de notificações iniciado");
            } else {
                System.out.println("Sistema já em execução");
            }
            return instanciaUnica;
        }

        // Adiciona uma nova notificação
        public void incluir(String mensagem) {
            filaNotificacoes.add(mensagem);
            System.out.println("Notificação registrada: " + mensagem);
        }

        // Exibe todas as notificações armazenadas
        public void exibir() {
            System.out.println("Lista de notificações:");
            for (String msg : filaNotificacoes) {
                System.out.println(msg);
            }
        }
    }

    // Programa principal
    public static void main(String[] args) throws IOException {
        BufferedReader leitor = new BufferedReader(new InputStreamReader(System.in));
        String primeiraEntrada = leitor.readLine();
        GerenciadorNotificacoes notificacoes = GerenciadorNotificacoes.acessar();

        String comando;
        while ((comando = leitor.readLine()) != null) {
            switch (comando) {
                case "nova_instancia":
                    GerenciadorNotificacoes.acessar();
                    break;
                case "adicionar":
                    notificacoes.incluir(primeiraEntrada);
                    break;
                case "listar":
                    notificacoes.exibir();
                    break;
                default:
                    System.out.println("Comando desconhecido: " + comando);
            }
        }
    }
}
